packages/core/core/src/TransformerRunner.js

```diff
@@ -136,28 +111,27 @@ class TransformerRunner {
       }
 
       // If the generated asset has the same type as the input...
-      if (result.type === context.type) {
+      // TODO: this is incorrect since multiple file types could map to the same pipeline. need to compare the pipelines.
```

Comment:
Could you give an example of when this might happen?

Reply:
i.e. in the default config, we have `*.{js,mjs,es6}` mapping to a single transformer pipeline. If someone had an `.mjs` file as input, but the output was always `js` from the transformer, then we'd get into an infinite loop here. So we should be comparing the pipelines, not the extensions.

Reply:
I guess I was thinking those would all still by type "js". Is type ever different than the extension?

Reply:
Well we initialize `type` to the file extension at the start of the pipeline: 
https://github.com/parcel-bundler/parcel/pull/2335/files#diff-e00a1c679d921332a0033bb784c5d4b7R54 This means that the input type might be `mjs` but the output of the transformer would be `js`. Currently the code would recurse and run the JS transformer again with an input type of `js` rather than stopping. So, we need to compare the pipeline rather than the type.